src/algebra/algebra/spectrum.lean

```diff
@@ -245,6 +243,88 @@ begin
     exact ⟨unit_mem_mul_iff_mem_swap_mul.mp k_mem, k_neq⟩ },
 end
 
+open polynomial
+/-- This is half of the spectral mapping theorem for polynomials. We prove it separately
+because it holds over any field, whereas `spectrum.polynomial_spectrum` needs the field to
+be algebraically closed. -/
+theorem polynomial_subset (a : A) (p : polynomial 𝕜) :
+  (λ k, eval k p) '' (σ a) ⊆ σ (aeval a p) :=
+begin
+  rintros _ ⟨k, hk, rfl⟩,
+  let q := C (eval k p) - p,
+  have hroot : is_root q k, by simp only [eval_C, eval_sub, sub_self, is_root.def],
+  rw [←mul_div_eq_iff_is_root, ←neg_mul_neg, neg_sub] at hroot,
+  have aeval_q_eq : ↑ₐ(eval k p) - aeval a p = aeval a q,
+    by simp only [aeval_C, alg_hom.map_sub, sub_left_inj],
+  rw [mem_iff, aeval_q_eq, ←hroot, aeval_mul],
+  have hcomm := aeval_comm a (C k - X) (- (q / (X - C k))),
+  apply mt (λ h, (hcomm.is_unit_mul_iff.mp h).1),
+  simpa only [aeval_X, aeval_C, alg_hom.map_sub] using hk,
+end
+
+lemma aeval_list_prod_is_unit {R M : Type*} [comm_semiring R] [ring M] [algebra R M] {m : M}
+  {s : list (polynomial R)} (h : ∀ p ∈ s, is_unit (aeval m p)) :
+  is_unit (aeval m s.prod) :=
+begin
+  induction s,
+  { simp only [aeval_one, is_unit_one, list.prod_nil] },
+  { have u_hd := h s_hd (list.mem_cons_self s_hd s_tl),
+    have u_tl := s_ih (λ p p_mem, h p (list.mem_of_mem_tail p_mem)),
+    rw [list.prod_cons, aeval_mul, (aeval_comm m _ _).is_unit_mul_iff],
+    exact ⟨u_hd, u_tl⟩ }
+end
+
+/-- This is the *spectral mapping theorem* for polynomials.  Note: the assumption `degree p > 0`
+is necessary in case `σ a = ∅`, for then the left-hand side is `∅` and the right-hand side,
+assuming `[nontrivial A]`, is `{k}` where `p = polynomial.C k`. -/
+theorem polynomial_eq_of_degree_pos [is_alg_closed 𝕜] (a : A) (p : polynomial 𝕜)
+  (hdeg : 0 < degree p) : (λ k, eval k p) '' (σ a) = σ (aeval a p) :=
```

Comment:
How about swapping the order of the equality? It seems like that would be the more usual direction for rewriting. Also, maybe `map_polynomial_eq_of_degree_pos` would be a better name: I think my first instinct would be to search for something like `spectrum.map` and hope that autocomplete will do the rest.

Reply:
I changed the name to `map_polynomial_of_degree_pos` (note: without `eq`). As for the order of rewriting, I'm not sure yet which is more natural, so we can go with your preference unless someone objects.